backend/pkg/kafka/config_helper.go

```diff
@@ -146,12 +147,37 @@ func NewKgoConfig(cfg *config.Kafka, logger *zap.Logger, hooks kgo.Hook) ([]kgo.
 
 		// AWS MSK IAM
 		if cfg.SASL.Mechanism == config.SASLMechanismAWSManagedStreamingIAM {
-			mechanism := aws.Auth{
-				AccessKey:    cfg.SASL.AWSMskIam.AccessKey,
-				SecretKey:    cfg.SASL.AWSMskIam.SecretKey,
-				SessionToken: cfg.SASL.AWSMskIam.SessionToken,
-				UserAgent:    cfg.SASL.AWSMskIam.UserAgent,
-			}.AsManagedStreamingIAMMechanism()
+			var mechanism sasl.Mechanism
+			if cfg.SASL.AWSMskIam.AccessKey != "" &&
+				cfg.SASL.AWSMskIam.SecretKey != "" {
+				// SessionToken is optional
+				mechanism = aws.Auth{
+					AccessKey:    cfg.SASL.AWSMskIam.AccessKey,
+					SecretKey:    cfg.SASL.AWSMskIam.SecretKey,
+					SessionToken: cfg.SASL.AWSMskIam.SessionToken,
+					UserAgent:    cfg.SASL.AWSMskIam.UserAgent,
+				}.AsManagedStreamingIAMMechanism()
+			} else {
+				cfgVal, err := awsConfig.LoadDefaultConfig(
+					context.TODO(),
```

Comment:
Worth to use a context with a reasonable timeout here I believe?

Reply:
updated to use a duration that is configurable for context. Unless someone is really working in a really crappy network, 10s is generous for getting credentials here.